mmv1/templates/terraform/metadata.yaml.tmpl

```diff
@@ -0,0 +1,5 @@
+resource: '{{ $.TerraformName }}'
```

Comment:
One thing that just crossed my mind: I wonder if we should take any steps to remove these files from the provider before it is released and/or built. I wouldn't expect them to impact the resulting build, but maybe we don't want them in the release source code archives.

Reply:
yes, we should remove them automatically if a TF resource is deleted, for example. I'm not sure offhand if this would happen automatically already.

Reply:
I was talking about the release artifacts here, but after another look, I don't think it's an issue to keep the yaml files.

For deleting these yaml files when the resources are removed, I confirmed this is covered under our current process (basically, `make clean-provider`)